bin/create_fortran_test.py

```diff
@@ -80,22 +81,27 @@ def write_testcase(c, tnum):
         expected = '.false.'
     else:
         expected = fix_and_quote_fortran_multiline(expected)
-    si.append('  ! Test %d: %s'%(tnum+1, description))
+    si.append('  ! Test %d: %s'%(TEST_NUMBER, description))
     if error:
         expected = 'ERROR'
         si.append('  ! ERROR: %s'%(error))
     si.append('  call assert_equal({}, {}, "{}")'.format(expected, inp, description))
     return si
 
-
+TEST_NUMBER=0
 def create_single_test(j):
     """Walk through the json cases and recursively write the test cases"""
     si = []
-    for tnum, c in enumerate(j['cases']):
+    global TEST_NUMBER
+    # there is a better way to do this by I dont have a good idea...
+    #print('A', TEST_NUMBER)
+    for c in j['cases']:
         if 'cases' in c:
             si.extend(create_single_test(c))
         else:
-            si.extend(write_testcase(c, tnum))
+            #print('B', TEST_NUMBER)
+            TEST_NUMBER = TEST_NUMBER +1
+            si.extend(write_testcase(c,TEST_NUMBER))
     return si
```

Comment:
To do this a little more cleanly we can flatten the test cases before handling them.

You're already flattening the test cases through recursion, and using the call stack as a data structure for managing it. You can convert the from using the call stack to using an actual stack

```python
si = []

nested_cases = j['cases']
flattened_cases = []

while nested_cases.len() > 0:
    cases = nested_cases.pop()
    if 'cases' in cases:
        map(nested_cases.append, cases['cases'])
    else:
        map(flattened_cases.append, cases)
        
for test_num, case in enumerate(flattened_cases):
    # etc
```

That's one option. Maybe better would be to have a function `flatten_test_cases` or something. (It also be flattened recursively of course)

Reply:
Very good idea, did just that :-)